app/src/main/java/com/mk/tasky/agenda/data/remote/sync/AgendaSyncImpl.kt

```diff
@@ -0,0 +1,23 @@
+package com.mk.tasky.agenda.data.remote.sync
+
+import androidx.work.*
+import com.mk.tasky.agenda.data.remote.worker.SyncWorker
+import com.mk.tasky.agenda.data.remote.worker.SyncWorker.Companion.WORKER_ID
+import com.mk.tasky.agenda.domain.sync.AgendaSync
+import java.time.Duration
+
+class AgendaSyncImpl(
+    private val workManager: WorkManager
+) : AgendaSync {
+    override suspend fun syncAgenda() {
+        val uploaderWorker = OneTimeWorkRequestBuilder<SyncWorker>().setConstraints(
+            Constraints.Builder().setRequiredNetworkType(NetworkType.CONNECTED).build()
+        ).setBackoffCriteria(BackoffPolicy.EXPONENTIAL, Duration.ofMinutes(1)).build()
+
+        workManager.beginUniqueWork(
```

Comment:
Shouldn't this be periodic work?

Reply:
You are right!
Gonna change it to periodic

Reply:
Not in this case since it syncs from DB to API
PeriodicWorker is useful to Sync API to DB